src/main/java/com/microsoft/cse/helium/app/services/keyvault/EnvironmentReader.java

```diff
@@ -0,0 +1,166 @@
+      System.exit(-1);
+    }
+
+    if (authType.equals(Constants.USE_MSI)) {
+      this.authType = Constants.USE_MSI;
+    } else if (authType.equals(Constants.USE_CLI)) {
+      this.authType = Constants.USE_CLI;
+    } else {
+      System.out.println("Usage: mvn clean spring-boot:run  "
+          + "-Dspring-boot.run.arguments=\"--h --authtype=<CLI|MSI> --kvname=<keyVaultName>\"");
+      System.exit(-1);
+    }
+  }
+
+  /**
+   * getAuthType.
+   *
+   * @return returns CLI or MSI.
+   */
+  @SuppressFBWarnings("DM_EXIT")
+  public String getAuthType() {
+    if (this.authType != null) {
+      return this.authType;
+    }
+
+    authType = System.getenv(Constants.AUTH_TYPE);
+
+    // If it is not set, use the MSI
+    if (authType == null) {
+      return Constants.USE_MSI;
+    }
+
+    // ONLY If it is set and values are either MSI or CLI, we will accept.
+    // otherwise, default is just the MSI
+    if (authType.equals(Constants.USE_MSI)) {
+      return Constants.USE_MSI;
+    } else if (authType.equals(Constants.USE_CLI)) {
+      return Constants.USE_CLI;
+    } else {
+      System.exit(-1);
+      // following return needs to be there because java compiler wants a return statement.
+      return null;
+    }
+  }
+
+  @SuppressFBWarnings("DM_EXIT")
+  private void setKeyVaultName(String kvName) {
+    if (kvName == null) {
+      System.out.println("Usage: mvn clean spring-boot:run  "
+          + "-Dspring-boot.run.arguments=\"--h --authtype=<CLI|MSI> --kvname=<keyVaultName>\"");
+      System.exit(-1);
+    }
+
+    if (!isValidKeyVaultName(kvName)) {
+      logger.error("helium.keyvault.name (KeyVaultName) value is '" + kvName
+          + "' which does not meet the criteria must be 3-24 characters long, begin with a "
+          + "character, may contain alphanumeric or hyphen, no repeating hyphens, and end with "
+          + "alphanumeric.  Check ${KeyVaultName} in your environment variables.");
+      System.out.println("Usage: mvn clean spring-boot:run"
+          + "-Dspring-boot.run.arguments=\"--authtype=<CLI|MSI> --kvname=<keyVaultName>\"");
+      System.exit(-1);
+    }
+
+    this.keyVaultName = kvName;
+  }
+
+  /**
+   * getKeyVaultName.
+   *
+   * @return returns the key vault name.
+   */
+  @SuppressFBWarnings("DM_EXIT")
+  public String getKeyVaultName() {
+    if (this.keyVaultName != null) {
+      return this.keyVaultName;
+    }
+
+    keyVaultName = System.getenv(Constants.KEY_VAULT_NAME);
+    if (keyVaultName == null || !isValidKeyVaultName(keyVaultName)) {
+      System.exit(-1);
+    }
+
+    return keyVaultName;
+  }
+
+  /**
+   * isValidKeyVaultName.
+   */
+  private Boolean isValidKeyVaultName(final String keyVaultName) {
+    Boolean validSetting = true;
+
+    if (keyVaultName.length() < 3 || keyVaultName.length() > 24) {
+      validSetting = false;
+    }
+
+    // validate key vault name with regex:
+    // ^[a-zA-Z](?!.*--)([a-zA-Z0-9-]*[a-zA-Z0-9])?$
```

Comment:
this needs to be reviewed so it's standard across languages

Reply:
@jofultz Do you remember where you picked this regex from?

Reply:
My brain :). I took the rules for keyvault names and used web tool to put together a regex for it.

Reply:
c# currently only does a IsNullOrWhiteSpace().  Maybe we can discuss at the end of stand-up whether or not to check the format or to pass on to KV API and let it handle the check.